### DISCUSSION TITLE
How can we uninstall the client silently?

### DISCUSSION THREAD

**Discussion by nicedevil007** [rustdesk/rustdesk-server-pro#260] [Q&A]
Heya, I deployed 1.2.4 on my clients and now I want to deploy my new application with all custom settings with a different appname. So first part of my script is to uninstall the old instance of rustdesk. Unfortunately I can't find the right Switch to turn of this window?

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/17103076/52220c4a-d36e-4b30-8b0e-cd03f401633c)

commands I tested so far are these here:

```powershell
--uninstall /q
--uninstall /SILENT
--uninstall /VERYSILENT
--uninstall /PASSIVE
--uninstall /qn
--uninstall /S
--uninstall /Q
--uninstall --silent
--uninstall --quiet
```

**Comment by rustdesk** [comment]
https://www.advancedinstaller.com/how-to-uninstall-msi-package.html

**Reply by nicedevil007** [reply]
The old installation isn't MSI install. It was installed with the EXE, does this make a difference ?

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/blob/bd7790c1eb17521cf6244e06a71a746fe60bba41/src/platform/windows.rs#L1288
https://github.com/rustdesk/rustdesk/blob/bd7790c1eb17521cf6244e06a71a746fe60bba41/src/platform/windows.rs#L1318

Try out below, I do not have a Windows right now, not test yet.
```
sc stop RustDesk
sc delete RustDesk
taskkill /F /IM RuntimeBroker_rustdesk.exe
taskkill /F /IM RustDesk.exe
reg delete HKEY_CLASSES_ROOT\.rustdesk /f
reg delete HKEY_CLASSES_ROOT\rustdesk /f
netsh advfirewall firewall delete rule name="RustDesk Service"
reg delete HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\RustDesk
del /f /q "C:\Program Files\RustDesk"
del /f /q "%ProgramData%\Microsoft\Windows\Start Menu\Programs\RustDesk"
del /f /q "%PUBLIC%\Desktop\RustDesk.lnk"
del /f /q "%PROGRAMDATA%\Microsoft\Windows\Start Menu\Programs\Startup\RustDesk Tray.lnk"
```

**Comment by fufesou** [comment]
It seems you have a MSI install.

## Non MSI install

`"C:\Program Files\RustDesk\rustdesk.exe" --uninstall` uninstall the product without a confirm window for version 1.2.4

## MSI install

`"C:\Program Files\RustDesk\rustdesk.exe" --uninstall` will show the same confirm window.

 `msiexec /x /qn <{ProductCode}>` works for me.

You can find the command in the register `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\RustDesk\QuietUninstallString`

<img width="1034" alt="1717209858392" src="https://github.com/rustdesk/rustdesk-server-pro/assets/13586388/4a526fff-882e-4a3a-beb7-1cdcbe4619a6">

**Reply by nicedevil007** [reply]
It is definitely not a MSI install. I found out that it was version 1.23.1 where the MSI option wasn't available.

And with just the --uninstall switch it is giving me the window that I want to avoid. Maybe I will do the complete manual way to remove it now.

Thank you all for the input!

**Comment by nicedevil007** [comment]
Because I use https://github.com/psappdeploytoolkit/psappdeploytoolkit I'm able to remove any MSI with the name RustDesk and can run the uninstall string from the registry. I will ignore anything of the errors and then I will do the cleanup from here posted by rustdesk themself.
